rotkehlchen/tests/db/test_history_events.py

```diff
@@ -605,3 +620,99 @@ def test_match_exact_events(database: 'DBHandler', start_with_valid_premium: boo
         assert result_match_grouped[0][0] == 2 and result_no_match_grouped[0][0] == 4
         assert result_match_grouped[0][1].asset == A_DAI
         assert result_match_grouped[0][1].asset == A_DAI
+
+
+@pytest.mark.parametrize('ethereum_accounts', [['0xf4ae64c5c4fb632D0e0D77097b957941c399d26e']])
+def test_deleting_events_by_location(database, ethereum_accounts):
+    """Test that deleting events by location does not remove EthWithdrawalEvent or EthBlockEvent instances.
```

Comment:
That's not the right test. That function does not matter. That function is meaningless. The bug was this function being called by the redecoding api logic.

Your test shoudl be an api test, and a test that makes sure that redecoding by the API does not delete those events.